server/game/Events/Event.js

```diff
@@ -93,10 +93,6 @@ class Event {
             this.cancel();
             return;
         }
-
-        if (this.card) {
```

Comment:
I am still not comfortable with this change.... There can be a situation, for example, simultaneous resolution, that this change could have broken.
I need to think about a situation and write a TC.